packages/keymanager-server/src/impl.ts

```diff
@@ -0,0 +1,226 @@
+import {SecretKey} from "@chainsafe/bls";
+import {Keystore} from "@chainsafe/bls-keystore";
+import {
+  Api,
+  DeletionStatus,
+  ImportStatus,
+  KeystoreStr,
+  SlashingProtectionData,
+} from "@chainsafe/lodestar-api/keymanager";
+import {fromHexString} from "@chainsafe/ssz";
+import {Interchange, ISlashingProtection, SecretKeyInfo, SignerType, Validator} from "@chainsafe/lodestar-validator";
+import {PubkeyHex} from "@chainsafe/lodestar-validator/src/types";
+import {Root} from "@chainsafe/lodestar-types";
+import {unlink, writeFile} from "fs/promises";
+import lockfile from "lockfile";
+import {ILogger} from "@chainsafe/lodestar-utils";
+
+export const LOCK_FILE_EXT = ".lock";
+
+export class KeymanagerApi implements Api {
+  constructor(
+    private readonly logger: ILogger,
+    private readonly validator: Validator,
+    private readonly slashingProtection: ISlashingProtection,
+    private readonly genesisValidatorRoot: Uint8Array | Root,
+    private readonly importKeystoresPath?: string[],
+    private readonly secretKeysInfo?: SecretKeyInfo[]
+  ) {}
+
+  /**
+   * List all validating pubkeys known to and decrypted by this keymanager binary
+   *
+   * https://github.com/ethereum/keymanager-APIs/blob/0c975dae2ac6053c8245ebdb6a9f27c2f114f407/keymanager-oapi.yaml
+   */
+  async listKeys(): Promise<{
+    data: {
+      validatingPubkey: PubkeyHex;
+      /** The derivation path (if present in the imported keystore) */
+      derivationPath?: string;
+      /** The key associated with this pubkey cannot be deleted from the API */
+      readonly?: boolean;
+    }[];
+  }> {
+    const pubkeys = this.validator.validatorStore.votingPubkeys();
+    return {
+      data: pubkeys.map((pubkey) => ({
+        validatingPubkey: pubkey,
+        derivationPath: "",
```

Comment:
Empty string is not a valid value. If derivationPath is not available, don't include in the JSON response. However the derivation path SHOULD be available since this is dealing with in disk keystores

Reply:
I read [EIP-2334](https://eips.ethereum.org/EIPS/eip-2334) over the weekend and I realised the changes I made based on this comment where I now set derivationPath to `m/12381/3600/0/0/0` is wrong, because the derivation path should be `m/12381/3600/i/0/0` where i indicates the ith set of validator keys. I do not see a quick way to get the i vallue with the current code, so as to make it possible to merge this PR, I suggest to have the deriviationPath set back to "", and then create a follow up where the proper `derivationPath` can be computed. Acceptable?